src/coreclr/vm/class.cpp

```diff
@@ -1222,6 +1222,17 @@ void ClassLoader::LoadExactParents(MethodTable* pMT)
     // We can now mark this type as having exact parents
     pMT->SetHasExactParent();
 
+#ifdef TARGET_RISCV64
+    if (pMT->HasLayout() &&
```

Comment:
It would be better to do this before `SetHasExactParent` call.

This is lock-free algorithm and `SetHasExactParent` acts as a barrier. If you do this after `SetHasExactParent`, it is possible for the other threads to run a lot further and hit the problem in theory.

Reply:
> It would be better to do this before `SetHasExactParent` call.

I thought about initializing native layout info earlier but that failed on ["Recursion in CLRException::GetThrowable"](https://github.com/dotnet/runtime/blob/4d69ab7d59e7a7ded3aaf56789740b0698f07b17/src/coreclr/vm/clrex.cpp#L132) so I committed after SetHasExactParent.

But if you say it's worth it, it's caused by a infinite loop in `EnsureNativeLayoutInfoInitialized()` loading ``System.ReadOnlySpan`1[__Canon]`` and `CoreLibBinder::GetClass(CLASS__INT128)` alternately. Not sure why after INT128 it comes back to load ReadOnlySpan, I'll try to find a circuit breaker.